### DISCUSSION TITLE
lifespan incompatible type

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [fastapi/fastapi#10415] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def lifespan(_: FastAPI) ->  AsyncGenerator[dict[str, str] | None, None]:
    async with CustomAsyncContextManager() as c:
	# startup stuff
	yield {'this' : that}
	# cleanup

app = FastAPI(lifespan=lifespan)	# L866
```

### Description

I'm using a custom async context manager with lifespan. The program runs as expected but mypy returns this error:

mypy:
`866: error: Argument "lifespan" to "FastAPI" has incompatible type "Callable[[FastAPI], AsyncGenerator[dict[str, Dao] | None, None]]"; expected "Callable[[FastAPI], AbstractAsyncContextManager[None]] | Callable[[FastAPI], AbstractAsyncContextManager[Mapping[str, Any]]] | None"  [arg-type]
`

What does this mean?

### Operating System

Windows

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

Python 3.11.4

**Comment by Kludex** [comment]
Duplicated https://github.com/tiangolo/fastapi/discussions/10413

**Reply by dineshbvadhia** [reply]
@Kludex Duplicate by mistake but won't accept deletion.

Is there an answer to the question? Thanks

**Reply by Kludex** [reply]
... It's duplicated. Please continue the conversation on the other discussion.
